app/lib/event_store/event.rb

```diff
@@ -64,6 +66,35 @@ def load(str)
 
     scope :stale, -> { where.has { created_at <= TTL.ago } }
 
+    class EventRollbackError < StandardError
+      include Bugsnag::MetaData
+
+      def initialize(error:, event:)
+        event_details = event.to_h.as_json
+        self.bugsnag_meta_data = { event: event_details }
+
+        super "Error raised trying to roll back #{event}: #{error}, event details: #{event_details}"
+        set_backtrace error.backtrace
```

Comment:
```suggestion
```

When the exception is raised, the `cause` is set. I absolutely hate exceptions that are not clear how and where they come from. I'll check how to generate it properly.

```
[4] pry(main)> class EEE < RuntimeError
[4] pry(main)*   def initialize(paramX)  
[4] pry(main)*     super "error in #{paramX}"    
[4] pry(main)*   end  
[4] pry(main)* end    

[11] pry(main)> begin
[11] pry(main)*   raise RuntimeError, "exception cause"  
[11] pry(main)* rescue  
[11] pry(main)*   raise EEE.new("paramXvalue")
[11] pry(main)* end  

EEE: error in paramXvalue
from (pry):32:in `rescue in __pry__'
Caused by RuntimeError: exception cause
from (pry):30:in `__pry__'
```